packages/components/src/data-model/word-query/error/word-query-error.js

```diff
@@ -14,34 +21,48 @@ export default class WordQueryError {
    * Creates an instance of WordQueryError.
    *
    * @param {string} message - An error message.
-   * @param {import('./word-query-error-codes.js').WordQueryErrorCodes} code - An error code.
-   *        It is required to match Apollo GraphQL convention for errors.
+   * @param {import('../../constants/error-codes.js').ErrorCodes} errCode - An error or a warning code.
    * @param {string[]} path - A path (a sequence of component names) where an error occurred.
    */
-  constructor (message, code, { path = [] } = {}) {
+  constructor (message, errCode, { origin, clAdapter, clAdapterMethod } = {}) {
     if (!message) {
-      throw new Error('WordQueryError requires a message to be provided')
+      throw new Error(this.constructor.errMsgs.NO_MESSAGE)
     }
-    if (!code) {
-      throw new Error('WordQueryError requires a code to be provided')
+    if (!errCode) {
+      throw new Error(this.constructor.errMsgs.NO_ERROR_CODE)
     }
     this.message = message
-    this.path = path
+    this.path = ['word']
```

Comment:
I've changed it to be more in line with the GraphQL specs. According to those, it should be a path within the GraphQL response to the elements that failed (see http://spec.graphql.org/draft/#example-90475 as an example). Since we do not have an exact element that failed (but may have that in the future) I decided to use the generic `word` element. No looking at the example more closely I realize that it would be better to use `homonyms` instead. I will do that change, if you don't mind.

Ideally we should identify the exact homonym and the lexeme that failed, but I'm not sure how easy will it be to do now.

Reply:
>I will do that change, if you don't mind.

ok thanks